core/src/main/java/io/fabric8/maven/core/service/kubernetes/JibBuildService.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.service.kubernetes;
+
+import io.fabric8.maven.core.service.BuildService;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.docker.util.ImageName;
+import io.fabric8.maven.docker.util.Logger;
+
+import java.util.List;
+import java.util.Objects;
+
+public class JibBuildService implements BuildService {
+
+    private BuildServiceConfig config;
+
+    private Logger log;
+    private JibBuildService() { }
+
+    public JibBuildService (BuildServiceConfig config, Logger log) {
+        Objects.requireNonNull(config, "config");
+        this.config = config;
+        this.log = log;
+    }
+
+    @Override
+    public void build(ImageConfiguration imageConfiguration) {
+       try {
+           BuildImageConfiguration buildImageConfiguration = imageConfiguration.getBuildConfiguration();
+           List<String> tags = buildImageConfiguration.getTags();
+
+           JibBuildConfiguration jibBuildConfiguration;
+           String fullName = "";
+           if (tags.size() > 0) {
+               for (String tag : tags) {
+                   if (tag != null) {
+                        fullName = new ImageName(imageConfiguration.getName(), tag).getFullName();
+                   }
+               }
+           } else {
+               fullName = new ImageName(imageConfiguration.getName(), null).getFullName();
+           }
+           log.debug("Image tagging succesfull!");
+           jibBuildConfiguration = JibBuildServiceUtil.getJibBuildConfiguration(config, buildImageConfiguration, fullName, log);
+           JibBuildServiceUtil.buildImage(jibBuildConfiguration, log);
+       } catch (Exception ex) {
+           throw new UnsupportedOperationException();
```

Comment:
Shouldn't you provide caught exception's message inside the exception you are throwing? Also, could you please explain why are you throwing `UnsupportedOperationException` here? Shouldn't this method be throwing `Fabric8ServiceException`